File: TransactionIngestor/Input/Ingestors/StdFmtJsonIngestor.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TransactionIngestor.Data;

namespace TransactionIngestor.Input.Ingestors
{
    public class StdFmtJsonIngestor : IIngestor
    {
        public string InputFileName
        {
            set;
            private get;
        }

        public IEnumerable<DataRecord> GetRecords()
        {
            using (FileStream fs = new FileStream(InputFileName, FileMode.Open, FileAccess.Read))
            using (StreamReader reader = new StreamReader(fs))
            using (JsonReader jsonReader = new JsonTextReader(reader))
            {
                while (jsonReader.Read())
                {
                    if (jsonReader.TokenType == JsonToken.StartObject)
                    {
                        JObject obj = JObject.Load(jsonReader);
                        DataRecord record = obj.ToObject<DataRecord>();
                        yield return record;
                    }
                }
            }
        }
    }
}

File: transactioningestor/input/ingestors/stdfmtjsoningestor.py
from transactioningestor.data import DataRecord
from transactioningestor.data import get_datarecords_from_file

class StandardFormatJsonIngestor(object):
    
    def __init__(self,inputfilename):
        self._inputfilename = inputfilename

    def get_elements(self):
        for record in get_datarecords_from_file(self._inputfilename):
            yield record
        